socorro/external/crashstorage_base.py

```diff
@@ -253,368 +242,65 @@ def remove(self, crash_id):
-        start_time = self.start_timer()
-        result = self.wrapped_crashstore.get_raw_crash(crash_id)
-        end_time = self.end_timer()
-        self.logger.debug("%s get_raw_crash %s", self.tag, end_time - start_time)
-        return result
-
-    def get_raw_dump(self, crash_id, name=None):
-        start_time = self.start_timer()
-        result = self.wrapped_crashstore.get_raw_dump(crash_id)
-        end_time = self.end_timer()
-        self.logger.debug("%s get_raw_dump %s", self.tag, end_time - start_time)
-        return result
+        try:
+            return self._raw_crash_data[crash_id]
+        except KeyError:
+            raise CrashIDNotFound(f"{crash_id} not found")
+
+    def get_raw_dump(self, crash_id, name):
+        try:
+            return self._dumps[crash_id][name]
+        except KeyError:
+            raise CrashIDNotFound(f"{crash_id} not found")
 
     def get_dumps(self, crash_id):
-        start_time = self.start_timer()
-        result = self.wrapped_crashstore.get_dumps(crash_id)
-        end_time = self.end_timer()
-        self.logger.debug("%s get_dumps %s", self.tag, end_time - start_time)
-        return result
+        try:
+            return self._dumps[crash_id]
+        except KeyError:
+            raise CrashIDNotFound(f"{crash_id} not found")
 
     def get_dumps_as_files(self, crash_id, tmpdir):
-        start_time = self.start_timer()
-        result = self.wrapped_crashstore.get_dumps_as_files(crash_id, tmpdir)
-        end_time = self.end_timer()
-        self.logger.debug("%s get_dumps_as_files %s", self.tag, end_time - start_time)
-        return result
-
-    def get_processed(self, crash_id):
-        start_time = self.start_timer()
-        result = self.wrapped_crashstore.get_processed(crash_id)
-        end_time = self.end_timer()
-        self.logger.debug("%s get_processed %s", self.tag, end_time - start_time)
-        return result
-
-    def remove(self, crash_id):
-        start_time = self.start_timer()
-        self.wrapped_crashstore.remove(crash_id)
-        end_time = self.end_timer()
-        self.logger.debug("%s remove %s", self.tag, end_time - start_time)
-
-
-class MetricsEnabledBase(RequiredConfig):
-    """Base class for capturing metrics for crashstorage classes"""
-
-    required_config = Namespace()
-    required_config.add_option(
-        "metrics_prefix",
-        doc="a string to be used as the prefix for metrics keys",
-        default="",
-    )
-    required_config.add_option(
-        "active_list",
-        default="save_processed_crash,act",
-        doc="a comma delimeted list of counters that are enabled",
-        from_string_converter=str_to_list,
-    )
-
-    def __init__(self, config, namespace=""):
-        self.config = config
-        self.metrics = markus.get_metrics(self.config.metrics_prefix)
-
-    def _make_key(self, *args):
-        return ".".join(x for x in args if x)
-
-
-class MetricsCounter(MetricsEnabledBase):
```

Comment:
I removed this. It's a faux crash storage that is used to keep a count of how many times the processor saves a crash report which is a proxy for how many crash reports the processor processed. It's better to add an incr to the relevant code in the processor than have all this faux crash storage code.